src/engine/ParsedRequestBuilder.cpp

```diff
@@ -80,9 +84,14 @@ void ParsedRequestBuilder::extractGraphStoreOperationDirect() {
   // cannot deduce the used protocol (http/https) from the raw HTTP request. We
   // default to `http` for the constructed IRIs.
   AD_CORRECTNESS_CHECK(std::holds_alternative<None>(parsedRequest_.operation_));
+  if (!host_.has_value()) {
+    throw std::runtime_error(
+        "Request for Graph Store Protocol with direct graph identification "
+        "requires the host header to be set.");
```

Comment:
```suggestion
        "requires the `host` header to be set.");
```
And can't this be an HTTP error with a suitable status code?